### ISSUE TITLE
LG G3 crashes when loading emulator

### ISSUE THREAD

**Issue by arafaelsales** [reicast/reicast-emulator#615]
Hello guys

I am not new to reicast, I used to play it on my Moto x,  but now, I am using a LG g3, snapdragon 801 w/ 2gb ram and I am unable to play any game using reicast. It simply force closes whenever I try to run the bios or any game and it takes me back to the phone menu. The thing is that I have been trying every single build and the only one that fully works is the r5, after this one, none works, even the one from Google play is not working.  And of course, r5 is out of date. I am using a stock rom on my phone, nothing is modified. 
Making a research, I found out that many other users are having this problem and it turns out that it is a qualcomm's problem, if it is so, there is nothing we can do right? Everything is totally perfect, folders, bios, setting, it is just this damn thing that it is happening,  I already downloaded the same build on Moto x and perfectly functional. What could be happening? Is there something that I can show to help you guys figure it out? If it is there anything, anything that I could do to help, I would be very grateful. 

Many thanks from Brazil 😑😊

**Comment by LoungeKatt** [comment]
Please don't use the same name as the previous issue when posting a new one that is different. It is hard to tell them apart.

If it is crashing, the best thing to do is open the application and cause a crash then open it again and immediately go to the menu and submit a log. This may also be the same issue that has already been discussed at https://github.com/reicast/reicast-emulator/issues/591

**Comment by arafaelsales** [comment]
whoaaa, really, and Sorry about the title, I'll check it next time.
So, about the logs, It generated 2, I up them to past bin, so here they are.

http://pastebin.com/8k5jK4qU (1427206904321.txt)
http://pastebin.com/DTjvuS2A (mem_alloc.txt)

Thank You once again guys

**Comment by skmp** [comment]
That's quite weird, I don't really see where the crash comes from... Hum.

**Comment by arafaelsales** [comment]
:( I dunno what to do, miss my reicast. 
😢

**Comment by skmp** [comment]
No worries, I'll look more into it. Have you tried one of the CI builds
from builds.reicast.com?

On Sun, Mar 29, 2015 at 6:41 AM, arafaelsales <notifications@github.com>
wrote:

> :( I dunno what to do, miss my reicast.
> [image: 😢]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reicast/reicast-emulator/issues/615#issuecomment-87357393>
> .
>



-- 
~skmp

**Comment by arafaelsales** [comment]
Hey,

Many of them, this build was the latest on from the Ci builds (by the time
I tried to run).
I have also tried Google play version, it doesn't work too.
The only one that really works is the R5 version.
Weird isn't it? The old version works, the current one doesn't.

**Comment by LoungeKatt** [comment]
The r5 version was before a change for another device that had the memory issues, I believe. Fixing that worked until Lollipop came along and moved things again. It seems to be a game of Jenga trying to find a place to insert the emulator.

**Comment by arafaelsales** [comment]
Exactly, once, I was told that this is a memory allocation issue, and there
was nothing that could be done, is that true?

Thanks

**Comment by LoungeKatt** [comment]
It depends. If there is a region of free memory big enough, it may be possible. If every region is too broken up, there isn't anything that can be done because the items taking up the space are usually part of the firmware. 
On an Xperia Play, there is absolutely nothing that can be done, but there is a better chance with more memory. Unlike a desktop, doing a defragment on a phone isn't really possible or at least wouldn't be permanent. It is up to the firmware to have grouped itself from the start.

**Comment by arafaelsales** [comment]
I see. Really thanks.
So, if there's nothing to do, we can close the issue.
😢

**Comment by LoungeKatt** [comment]
We can leave it open for now, but I'll tag it.

**Comment by XSEGABOX** [comment]
Hello! Thank you all for this wonderful emulator. I have the same issue as @arafaelsales! Please do not close this yet as I hope to see a solution here one day! Reicast r5 works as it should, all other builds however do not. :( I have the same device LG G3 but odel is Korean F400L and with lollipop. My variant has 3 Gigs of ram...It is curious why r5 works and r5+ don't...Developer please help us. Thank you

**Comment by arafaelsales** [comment]
Yaaayyyyy can't believe someone's sharing the same issue with me. Now guys,
I really believe that this should be fixed soon, if there's anything,
anything that we could do to help you searching for this problem, pls, let
us know.
Weird is knowing that a version that is out of date works. Hahahaha

**Comment by LoungeKatt** [comment]
@XSEGABOX if you have the same crash, but a different device, don't forget to post your log. We can't try to locate an open region of memory without seeing what's available and that log will list it.

**Comment by arafaelsales** [comment]
@TwistedUmbrella when you say memory, you refer to ram or storage? Coz both
devices have high memories, international version has 2gb ram, like mine,
and the other 3gb ram.

**Comment by skmp** [comment]
The problem is not with the actual memory available, the problem is that we
need a continuous memory space. Almost all devices have enough ram, but
often it is "fragmented" with various libraries.

On Mon, Mar 30, 2015 at 5:59 PM, arafaelsales <notifications@github.com>
wrote:

> @TwistedUmbrella when you say memory, you refer to ram or storage? Coz both
> devices have high memories, international version has 2gb ram, like mine,
> and the other 3gb ram.
> On 30 Mar 2015 12:45, "TwistedUmbrella" <notifications@github.com> wrote:
>
> > @XSEGABOX <https://github.com/XSEGABOX> if you have the same crash, but
> a
>
> > different device, don't forget to post your log. We can't try to locate
> an
> > open region of memory without seeing what's available and that log will
> > list it.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
> https://github.com/reicast/reicast-emulator/issues/615#issuecomment-87728603
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/reicast/reicast-emulator/issues/615#issuecomment-87734978>
> .
>



-- 
~skmp

**Comment by LoungeKatt** [comment]
^ This. That is why defragment would be great for Android, but it isn't possible. Instead, it is a game of musical chairs trying to find a large enough range in that mem_alloc that won't block or get blocked by something important.

**Comment by arafaelsales** [comment]
So we know that it might be due to how the rom (android) allocate memory in
itself, right now, I am using a stock rom, maybe if I use cyanomod gen or
aosp, maybe it works, what do ya think?

**Comment by vgturtle127** [comment]
@arafaelsales When he says memory, he means RAM. Memory has never meant storage, but people kept saying memory and it became commonplace. Storage is long-term storage, RAM is short-term (generally speaking).

In regards to your statement above, it doesn't have to do with what ROM you are using. When they are talking about memory allocation, they mean that the system is trying to set aside RAM so it can be used by the application. Changing ROM's or even Android versions might not necessarily make a difference.

Although I had a similar experience with Dolphin, so you may try restarting your phone and then uninstalling the app and deleting all Reicast files and then try r6 or a newer CI build. I did that with Dolphin and it fixed the weird crashing I was having, and I was using an LG G3 at the time (but maybe this isn't a similar problem, but you could try it).

**Comment by arafaelsales** [comment]
Hey you,

Thanks in advance for the explanation. About build, I did try, last build I
tried was from March 27th.
I can't try newer ones, but I'm quite sure it won't work, as usual.

**Comment by arafaelsales** [comment]
Edit * I can try it, but I'm quite sure it won't work.

**Comment by jose325i** [comment]
I have an lg g2 that just updated to stock lollipop 5.0.2. Reicast used to work but now it does this where the app just closes once I open a game.
Here is a paste:
http://loungekatt.no-ip.biz:3194/ReicastBot/report/logs/1427843205387.txt

**Comment by arafaelsales** [comment]
We're all sharing the same issue, hope they find a way to overcome this.

**Comment by vgturtle127** [comment]
Maybe it has something to do with the recent upgrade to Lollipop?

@arafaelsales are you on Lollipop?

**Comment by arafaelsales** [comment]
Hey,

Yes, I am, stock 5.0 lollipop.

**Comment by vgturtle127** [comment]
Is it 5.0 or 5.0.2?

**Comment by arafaelsales** [comment]
5.0

**Comment by vgturtle127** [comment]
Maybe try 5.0.2? They were bug fix releases.

**Comment by arafaelsales** [comment]
Yeah, I know, but 5.0.2 releases are all aosp roms, they ain't good cause
they are generic, at least that's what I was told, there's a community
about LG g3 here in Brasil at Google plus, I'll stick to them asking
someone to test it.

**Comment by vgturtle127** [comment]
I guess, but Android has always been generic enough that it works on everything. Especially device like the LG G3 and Samsung devices and flagship phones like that.

Whatever is fine though. We still need to figure out what is causing the problem, though.

**Comment by skmp** [comment]
Just for reference, I just installed and tested latest reicast,
#ebabdee, and it boots and runs ikaruga (chd) just fine on my nexus 4
under 5.0.1.

**Comment by arafaelsales** [comment]
Baahhh, you are using the cleanest and the clearest Android phone in the
world.
Just like I told you, Moto G also runs fine.

**Comment by jose325i** [comment]
Did a factory reset since my phone was having issues and installed the ebabdee build. Same result. I was so excited to see if the new adreno v88 drivers and 5.0.2 fixed my stuttering issue introduced in 4.4.2. Turns out now I can't even boot anything haha
http://loungekatt.no-ip.biz:3194/ReicastBot/report/logs/1427865603569.txt

**Comment by LoungeKatt** [comment]
@jose325i If you have a section in the log labeled Memory Allocation Table (which you do) there is nothing that can be done on your end to fix it because some of the region reicast uses is already being used by the firmware.

**Comment by LoungeKatt** [comment]
Just so everyone is on the same page:

* Reicast needs a continuous region of available virtual memory to allocate
  * Reicast specifies the region by memory addresses
  * The region is not detected, it is hardcoded into the emulator

* Allocation is a first come, first served process with the firmware being first
  * Different Android versions allocate things to different regions
  * RAM size does not matter if the allocations are too fragemented
  * Factory resets and modified ROMs of the same version do not change allocation

* Different build types may change the allocation table
  * Must be built from source, not modified from an existing build
  * Must modify the native code for system libraries

**Comment by skmp** [comment]
Actually, reicast does not use any specific address, it asks the OS for
one. We need 512+32 megs of continuous address space, and commit around 30
megs.
Am 01.04.2015 16:01 schrieb "TwistedUmbrella" <notifications@github.com>:

**Comment by skmp** [comment]
Since I am in a really good mood today I will write a fallback. It will be
40% slower on affected phones, but new hardware can take that and still run
fast enough....
Am 02.04.2015 00:20 schrieb "skmp" <skmp@emudev.org>:

**Comment by arafaelsales** [comment]
Woooow, can't believe it. Really thanks.
😇😇😇😇😇😇😇😇😇😇

**Comment by LoungeKatt** [comment]
@skmp I thought you had it looking in a set range? At least with Xperia, libjpeg was where reicast wanted to go. I'll change that part.

**Comment by jose325i** [comment]
Just tested r5 and I can confirm that it works fine on my 5.0.2 g2. Perhaps an easier way to get around this would be to have the emulator change behavior back to whatever r5 does as a fallback if the current method fails?

**Comment by arafaelsales** [comment]
That's what he is going to do for us
😇😇😇😇😇😇

**Comment by vgturtle127** [comment]
@LoungeKatt Oh, I guess I was incorrect then.

OK, so basically it is solely based on the specific Android version that whatever ROM you are using is based on (including stock vanilla Android).

So any ROM based off of 5.0.2 would be the same as another ROM based off 5.0.2. That makes sense.

**Comment by LoungeKatt** [comment]
@vgturtle127 No worries. It is extremely confusing and there are too many words that can mean too many things, haha. I only know because I went around in circles back with the XPlay. I kept thinking I could modify the ROM but ended up wasting a lot of time getting nowhere.

**Comment by vgturtle127** [comment]
"wasting my time getting nowhere"

Just like iOS development! :laughing:

**Comment by LoungeKatt** [comment]
Lol. Yeah... They denied an update to an app because "it doesn't meet our standards for interface layout" when the all that was updated was the backend and services. The UI was a replica of a video game and ironically the same one as the last two versions they approved.

**Comment by vgturtle127** [comment]
Oh Apple, you card.

**Comment by LoungeKatt** [comment]
Add https://github.com/reicast/reicast-emulator/issues/627 to the list

**Comment by skmp** [comment]
Try e2a70294d76de4b02750337e1dc684b8aa7c0100 http://reicast-builds.s3.amazonaws.com/builds/heads/feat/no-direct-memmap-e2a7029/reicast-android-debug-e2a7029.apk, It doesn't use the memory optimization

**Comment by jose325i** [comment]
^This build fixes the issue for me. Games boot normally and work great. Definitely comes with a noticeable performance hit (Soul Calibur drops to 30-40 fps sometimes), but it's still playable. Thanks!

Sonic adventure 2 is really broken on this build for some reason though. Audio skips like crazy, and gameplay is super fast and choppy. It wasn't like this on r6 (before I got lollipop). Really weird.

**Comment by skmp** [comment]
Thats a separate issue (it's fixed on another branch)
Am 09.05.2015 23:16 schrieb "jose325i" <notifications@github.com>:

**Comment by cmexp** [comment]
I can confirm that the linked build works for me on my Lollipop HTC One M8 after having this issue on previous builds.

**Comment by skmp** [comment]
Closing this as it is quite old. Is it still an issue in current master?
